internal/builders/bazel/build.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -eu
+#
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+mkdir binaries
+
+# Transfer flags and targets to their respective arrays
+IFS=' ' read -r -a BUILD_FLAGS <<< "${FLAGS}"
+IFS=' ' read -r -a BUILD_TARGETS <<< "${TARGETS}"
+
+# Build with respect to entire arrays of flags and targets
+bazel build "${BUILD_FLAGS[@]}" "${BUILD_TARGETS[@]}"
+
+# Using target string, copy artifact to binaries dir
+for CURR_TARGET in "${BUILD_TARGETS[@]}"; do
+  CD_PATH=${CURR_TARGET%:*}
+  CD_PATH=${CD_PATH////}
+  BINARY_NAME=${CURR_TARGET#*:}
+  cp "bazel-bin/$CD_PATH/$BINARY_NAME" ./binaries
+done
```

Comment:
Does this allow us do use other rules packages to build artifacts (e.g. [rules_python](https://github.com/bazelbuild/rules_python))? Do output artifacts always get picked up this way?

I suspect that most popular rules are sufficiently hermetic and just generate files without any side-effects but I'm not 100% sure.

Reply:
If not, it would be nice to update the docs to include which types of rules we support and which we don't.

Reply:
I am unsure about other rule packages, and I will look into it further and update the readme with my findings!

Reply:
I think it should support the other rules, but will probably have to test

Reply:
I tested the python rule set and it will be supported when the runfile feature gets added. Runfile feature is being developed currently and is coming soon.

Reply:
Cool! I look forward to seeing some docs and examples in the future!